packages/compass-crud/src/components/insert-json-document.tsx

```diff
@@ -23,34 +34,86 @@ type InsertJsonDocumentProps = {
   updateJsonDoc: (value: string) => void;
 };
 
+function blankLeafValues(jsonDoc: string) {
+  return (
+    jsonDoc
+      // strings
+      .replace(/: "[^"]+"/g, ': ""')
+      // numbers
+      .replace(/: -?\d+(\.\d+)?/g, ': ')
+      // booleans
+      .replace(/: (true|false)/g, ': ')
+  );
+}
+
+function getInitialText(jsonDoc: string, isCommentNeeded: boolean) {
+  return isCommentNeeded ? `${EDITOR_COMMENT}${jsonDoc}` : jsonDoc;
+}
+
 const InsertJsonDocument: React.FunctionComponent<InsertJsonDocumentProps> = ({
   darkMode,
   jsonDoc,
   isCommentNeeded,
   updateJsonDoc,
 }) => {
   const [text, setText] = useState(() => {
-    return isCommentNeeded ? `${EDITOR_COMMENT}${jsonDoc}` : jsonDoc;
+    return getInitialText(jsonDoc, isCommentNeeded);
+  });
+
+  const [isBlank, setIsBlank] = useState(false);
+  const [blankText, setBlankText] = useState(() => {
+    return blankLeafValues(getInitialText(jsonDoc, isCommentNeeded));
   });
+  const [isChanged, setIsChanged] = useState(false);
 
   const onChangeText = (value: string) => {
-    setText(value);
     updateJsonDoc(value.split('*/\n').pop() ?? '');
+
+    // Don't disable the With Field Values checkbox unless the user changed the
+    // text.
+    if (value === blankText || value === text) {
+      return;
+    }
+
+    setIsChanged(true);
+    setText(value);
   };
 
+  const onChangeBlank = () => {
+    if (isBlank) {
+      setIsBlank(false);
+    } else {
+      setBlankText(blankLeafValues(text));
+      setIsBlank(true);
+    }
+  };
+
+  const isDisabled = isChanged;
+
   return (
-    <div
-      className={cx(
-        darkMode ? editorContainerStylesDark : editorContainerStylesLight
-      )}
-    >
-      <CodemirrorMultilineEditor
-        data-testid="insert-document-json-editor"
-        language="json"
-        text={text}
-        onChangeText={onChangeText}
-        initialJSONFoldAll={false}
-        minLines={18}
+    <div>
+      <div
+        className={cx(
+          darkMode ? editorContainerStylesDark : editorContainerStylesLight
+        )}
+      >
+        <CodemirrorMultilineEditor
+          data-testid="insert-document-json-editor"
+          language="json"
+          text={isBlank ? blankText : text}
+          onChangeText={onChangeText}
+          initialJSONFoldAll={false}
+          minLines={18}
+        />
+      </div>
+      <Checkbox
+        className={checkboxStyles}
+        data-testid="with-field-values"
+        onChange={onChangeBlank}
+        label="With Field Values"
```

Comment:
How is a user expected to know what “With Field Values” means? Without playing around with it, I would expect it to modify the behavior of the insertion itself, not the editor above it. I see the conversation in the linked ticket about this, but this just really seems like bad UX, and if we do go with a checkbox here, we should really adjust the phrasing to be better.